extensions/amp-tiktok/0.1/amp-tiktok.js

```diff
@@ -0,0 +1,253 @@
+              fill="black"
+            />
+            <path
+              d="M38.9726 11.69V10.1468C37.0728 10.1497 35.2104 9.61856 33.5986 8.61416C35.0254 10.1736 36.9042 11.2489 38.9726 11.69ZM28.962 1.76512C28.9136 1.48918 28.8765 1.21143 28.8507 0.932611V0H21.545V28.9609C21.5333 32.3367 18.7888 35.0704 15.4039 35.0704C14.4101 35.0704 13.4718 34.8349 12.6409 34.4165C13.7631 35.8866 15.535 36.8356 17.5296 36.8356C20.9139 36.8356 23.659 34.1021 23.6708 30.7259V1.76512H28.962ZM17.268 17.3268V15.6839C16.6575 15.6006 16.0421 15.5588 15.4259 15.5591C8.01082 15.5589 2 21.5594 2 28.9609C2 33.6013 4.36236 37.6908 7.95215 40.0964C5.58448 37.6803 4.12567 34.3733 4.12567 30.7257C4.12567 23.4186 9.98365 17.477 17.268 17.3268Z"
+              fill="#00F2EA"
+            />
+          </g>
+          <defs>
+            <clipPath id="clip0">
+              <rect width="44" height="44" fill="white" />
+            </clipPath>
+          </defs>
+        </svg>
+      `,
+    };
+  }
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {!Element} */
+    this.iframe_ = null;
+
+    /** @private {string} */
+    this.videoId_ = null;
+
+    /** @private {?Function}*/
+    this.unlistenMessage_ = null;
+
+    /** @private {?Promise} */
+    this.resolveRecievedFirstMessage_ = null;
+
+    /** @private {string} */
+    this.iframeNameString_ = this.getIframeNameString_();
+
+    this.resizeOuter_ = debounce(
+      this.win,
+      (height) => {
+        resetStyles(this.iframe_, [
+          'width',
+          'height',
+          'position',
+          'opacity',
+          'pointer-events',
+        ]);
+        this.iframe_.removeAttribute('aria-hidden');
+        this.iframe_.setAttribute('aria-label', 'Tiktok');
+        this.iframe_.classList.remove('i-amphtml-tiktok-unresolved');
+        this.iframe_.classList.add('i-amphtml-tiktok-centered');
+        this.forceChangeHeight(height);
+      },
+      1000
+    );
+  }
+
+  /**
+   * @param {boolean=} opt_onLayout
+   * @override
+   */
+  preconnectCallback(opt_onLayout) {
+    //See
+    //https://developers.tiktok.com/doc/Embed
+    Services.preconnectFor(this.win).url(
+      this.getAmpDoc(),
+      'https://www.tiktok.com',
+      opt_onLayout
+    );
+  }
+
+  /** @override */
+  buildCallback() {
+    const {src} = this.element.dataset;
+    const videoIdRegex = /^((.+\/)?)(\d+)\/?$/;
+    if (src) {
+      this.videoId_ = src.replace(videoIdRegex, (_, _1, _2, id) => id);
```

Comment:
Nit: Move the definition within this if statement since it's not used within the outer scope of the `buildCallback()` method.